components/blitz/src/omero/model/SmartShape.java

```diff
@@ -214,7 +218,50 @@ public static void pointsByBoundingBox(Shape s, Rectangle2D r,
                     }
                 }
             }
+        }
+
+        public static java.awt.geom.AffineTransform getAwtTransform(AffineTransform transform) {
+            if (transform == null) return null;
+            return 
+                new java.awt.geom.AffineTransform(
+                    transform.getA00().getValue(),
+                    transform.getA10().getValue(),
+                    transform.getA01().getValue(),
+                    transform.getA11().getValue(),
+                    transform.getA02().getValue(),
+                    transform.getA12().getValue());
+        }
 
+        public static Shape transformAwtShape(Shape shape, AffineTransform transform) {
+            if (transform == null) return shape;
+            final java.awt.geom.AffineTransform t = 
+                SmartShape.Util.getAwtTransform(transform);
+            return t.createTransformedShape(shape);
+        }
+        
+        public static Set<Point2D> getQuantizedLinePoints(Line2D line, Set<Point2D> points) {
+            if (line == null) return null;
+            
+            Point2D start = line.getP1();
+            Point2D end = line.getP2();
+            Point2D m = new Point2D.Double(
+                end.getX()-start.getX(), end.getY()-start.getY());
+            double lengthM = (Math.sqrt(m.getX()*m.getX()+m.getY()*m.getY()));
```

Comment:
if ``start == end == (0, 0) then lengthM == 0``

Reply:
I have addressed that degenerate case of a line being a point really with my latest commit. not sure why github doesn't mark it as outdated...